File: 0056_Merge Intervals/MergeIntervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        if(intervals == null || intervals.length == 0)
            return intervals;
        
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        
        LinkedList<int[]> list = new LinkedList<>();
        for(int[] curr : intervals) {
            if(list.isEmpty() || list.getLast()[1] < curr[0])
                list.add(curr);
            else 
                list.getLast()[1] = Math.max(list.getLast()[1], curr[1]);
        }
        
        return list.toArray(new int[list.size()][]);
    }
}

File: 0056_Merge Intervals/MergeIntervals.cs
using System;
using System.Collections.Generic;
using System.Linq;
public class Solution
{
    public int[][] Merge(int[][] intervals)
    {
        Array.Sort(intervals, (a, b) => a[0] - b[0]);
        var list = new LinkedList<int[]>();

        foreach (var interval in intervals)
        {
            if(list.Count == 0 || list.Last.Value[1] < interval[0])
            {
                list.AddLast(interval);
            }
            else
            {
                list.Last.Value[1] = Math.Max(list.Last.Value[1], interval[1]);
            }
        }

        return list.ToArray();
    }
}
